modules/checkbox/react/spec/Checkbox.spec.tsx

```diff
@@ -1,102 +1,134 @@
 import * as React from 'react';
-import {mount} from 'enzyme';
+import {render, fireEvent} from '@testing-library/react';
 import Checkbox from '../lib/Checkbox';
 import ReactDOMServer from 'react-dom/server';
 import {axe} from 'jest-axe';
 import FormField from '@workday/canvas-kit-react-form-field';
 
+const label = 'Test Checkbox';
+
 describe('Checkbox', () => {
   const cb = jest.fn();
   afterEach(() => {
     cb.mockReset();
   });
 
-  test('render an checkbox input with id', () => {
-    const component = mount(<Checkbox id="myCheckbox" onChange={cb} />);
-    expect(component.find('input').props().id).toBe('myCheckbox');
-    component.unmount();
-  });
+  describe('when rendered', () => {
+    describe('with an id', () => {
+      test('should render a checkbox input with id', () => {
+        const id = 'myCheckbox';
+        const {getByLabelText} = render(<Checkbox id={id} onChange={cb} label={label} />);
+        expect(getByLabelText(label)).toHaveAttribute('id', id);
+      });
+    });
 
-  test('render an checkbox input with value', () => {
-    const component = mount(<Checkbox value="myCheckbox" onChange={cb} />);
-    expect(component.find('input').props().value).toBe('myCheckbox');
-    component.unmount();
-  });
+    describe('with a value', () => {
+      test('should render a checkbox input with value', () => {
+        const value = 'myCheckbox';
+        const {getByDisplayValue} = render(<Checkbox value={value} onChange={cb} />);
+        expect(getByDisplayValue(value)).toBeDefined();
+      });
+    });
 
-  test('render an checked checkbox input', () => {
-    const component = mount(<Checkbox checked={true} onChange={cb} />);
-    expect(component.find('input').props().checked).toBe(true);
-    component.unmount();
-  });
+    describe('with checked=true', () => {
+      test('should render a checked checkbox input', () => {
+        const {getByLabelText} = render(<Checkbox checked={true} onChange={cb} label={label} />);
+        expect(getByLabelText(label)).toHaveProperty('checked', true);
```

Comment:
Would love to use the new `toHaveChecked` matcher from `@testing-library/jest-dom`, but there seem to be type issues within Emotion. We can update these when we update to Emotion 11 (see https://github.com/emotion-js/emotion/issues/1689 for more info)